registry/session.go

```diff
@@ -281,7 +281,11 @@ func (r *Session) GetRepositoryData(remote string) (*RepositoryData, error) {
 	// TODO: Right now we're ignoring checksums in the response body.
 	// In the future, we need to use them to check image validity.
 	if res.StatusCode != 200 {
-		return nil, utils.NewHTTPRequestError(fmt.Sprintf("HTTP code: %d", res.StatusCode), res)
+		errBody, err := ioutil.ReadAll(res.Body)
+		if err != nil || len(errBody) == 0 {
```

Comment:
The error is essentially being ignored in this case. If the err is only informational for not used by the caller, then it should just be logged and only one return should occur.

Essentially
~~~
if err != nil  {
	log.Debugf("Error reading response body: %s", err)
}
return nil, utils.NewHTTPRequestError(fmt.Sprintf("Error: Status %d trying to pull repository %s: %s", res.StatusCode, remote, errBody), res)
~~~